Could not compile `pbr`  -- macOs Sierra 10.12.4

```bash
rust-youtube-downloader git:(master) rustc -V
rustc 1.17.0-nightly (8c4f2c64c 2017-03-22)
rust-youtube-downloader git:(master) cargo run -- l6zpi90IT1g
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading pbr v0.3.1
 Downloading hyper-native-tls v0.2.2
 Downloading hyper v0.10.6
 Downloading native-tls v0.1.2
 Downloading antidote v1.0.0
 Downloading traitobject v0.1.0
 Downloading rustc-serialize v0.3.23
 Downloading idna v0.1.1
 Downloading security-framework v0.1.14
 Downloading security-framework-sys v0.1.14
   Compiling language-tags v0.2.2
   Compiling libc v0.2.21
   Compiling httparse v1.2.1
   Compiling traitobject v0.1.0
   Compiling winapi-build v0.1.1
   Compiling core-foundation-sys v0.2.3
   Compiling antidote v1.0.0
   Compiling rustc-serialize v0.3.23
   Compiling num_cpus v1.3.0
   Compiling matches v0.1.4
   Compiling log v0.3.7
   Compiling winapi v0.2.8
   Compiling core-foundation v0.2.3
   Compiling time v0.1.36
   Compiling unicode-normalization v0.1.4
   Compiling security-framework-sys v0.1.14
   Compiling mime v0.2.3
   Compiling semver v0.1.20
   Compiling rand v0.3.15
   Compiling security-framework v0.1.14
   Compiling rustc_version v0.1.7
   Compiling unicase v1.4.0
   Compiling hyper v0.10.6
   Compiling tempdir v0.3.5
   Compiling kernel32-sys v0.2.2
   Compiling unicode-bidi v0.2.5
   Compiling pbr v0.3.1
error[E0283]: type annotations required: cannot resolve `std::string::String: std::convert::AsRef<_>`
   --> /Users/hon/.cargo/registry/src/github.com-1ecc6299db9ec823/pbr-0.3.1/src/pb.rs:348:64
    |
348 |                         base = base + repeat!(self.bar_current.as_ref(), curr_count - 1) +
    |                                       -------------------------^^^^^^------------------- in this macro invocation

error: aborting due to previous error

error: Could not compile `pbr`.
Build failed, waiting for other jobs to finish...
error: build failed
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,8 @@
+language: rust
+rust:
+  - stable
+  - beta
+  - nightly
+matrix:
+  allow_failures:
+    - rust: nightly

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -4,8 +4,8 @@ version = "0.1.0"
 authors = ["smoqadam <phpro.ir@gmail.com>"]
 
 [dependencies]
-hyper="*"
-hyper-native-tls="*"
-pbr="*"
-clap="*"
-regex="*"
+hyper="0.10.8"
+hyper-native-tls="0.2.2"
+pbr = "1.0.0-alpha.3"
+clap="2.23.2"
+regex="0.2.1"

--- a/README.md
+++ b/README.md
@@ -2,6 +2,8 @@
 
 YouTube video downloader written in Rust.
 
+[![Build Status](https://travis-ci.org/smoqadam/rust-youtube-downloader.svg?branch=master)](https://travis-ci.org/smoqadam/rust-youtube-downloader)
+
 ## How to use
 
 ```bash

--- a/README.md
+++ b/README.md
@@ -1,5 +1,6 @@
-## Rust Youtube Downloader
-Youtube video downloader written in Rust
+## Rust YouTube Downloader
+
+YouTube video downloader written in Rust.
 
 ## How to use
 
@@ -9,11 +10,11 @@ $ cd rust-youtube-downloader
 $ cargo run -- [youtube video id or youtube URL]
 ```
 
-for example:
+For example:
 
 `$ cargo run -- l6zpi90IT1g`
 
 ## Contributing
-this project is making for learning purposes and maybe/surly it has a lot of bugs and needs more improvement. Please let me know if you find any bugs by opening an issue or send a PR.
 
+This project is for learning purposes and may contain bugs. Please let me know if you find any by opening an issue or send a PR.
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -72,8 +72,8 @@ fn download(url: &str) {
                  quality["type"]);
     }
 
-    println!("Choose quality: ");
-    let input = read_line().trim().parse().unwrap();
+    println!("Choose quality (0): ");
+    let input = read_line().trim().parse().unwrap_or(0);
 
     println!("Please wait...");
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -42,6 +42,12 @@ fn download(url: &str) {
     let mut response_str = String::new();
     response.read_to_string(&mut response_str).unwrap();
     let hq = parse_url(&response_str);
+
+    if hq["status"] != "ok" {
+        println!("Video not found!");
+        return;
+    }
+
     // get video info
     let streams: Vec<&str> = hq.get("url_encoded_fmt_stream_map")
         .unwrap()

--- a/src/main.rs
+++ b/src/main.rs
@@ -130,10 +130,7 @@ fn send_request(url: &str) -> Response {
     let ssl = NativeTlsClient::new().unwrap();
     let connector = HttpsConnector::new(ssl);
     let client = Client::with_connector(connector);
-    match client.get(url).send() {
-        Ok(response) => response,
-        Err(why) => panic!("{}", why),
-    }
+    client.get(url).send().unwrap()
 }
 
 fn parse_url(query: &str) -> HashMap<String, String> {

--- a/src/main.rs
+++ b/src/main.rs
@@ -49,36 +49,27 @@ fn download(url: &str) {
     }
 
     // get video info
-    let streams: Vec<&str> = hq.get("url_encoded_fmt_stream_map")
-        .unwrap()
-        .split(",")
+    let streams: Vec<&str> = hq["url_encoded_fmt_stream_map"]
+        .split(',')
         .collect();
-
-    // get video title
-    let title = hq.get("title").unwrap();
-
-    let mut i = 0;
 
     // list of available qualities
     let mut qualities: HashMap<i32, (String, String)> = HashMap::new();
-    for url in streams.iter() {
-        i += 1;
-        let quality = parse_url(&url);
-        let extension = quality
-            .get("type")
-            .unwrap()
-            .split("/")
+    for (i, url) in streams.iter().enumerate() {
+        let quality = parse_url(url);
+        let extension = quality["type"]
+            .split('/')
             .nth(1)
             .unwrap()
-            .split(";")
+            .split(';')
             .next()
             .unwrap();
-        qualities.insert(i,
-                         (quality.get("url").unwrap().to_string(), extension.to_owned()));
+        qualities.insert(i as i32,
+                         (quality["url"].to_string(), extension.to_owned()));
         println!("{}- {} {}",
                  i,
-                 quality.get("quality").unwrap(),
-                 quality.get("type").unwrap());
+                 quality["quality"],
+                 quality["type"]);
     }
 
     println!("Choose quality: ");
@@ -86,8 +77,8 @@ fn download(url: &str) {
 
     println!("Please wait...");
 
-    let url = &qualities.get(&input).unwrap().0;
-    let extension = &qualities.get(&input).unwrap().1;
+    let url = &qualities[&input].0;
+    let extension = &qualities[&input].1;
 
     // get response from selected quality
     let response = send_request(url);
@@ -96,7 +87,7 @@ fn download(url: &str) {
     // get file size from Content-Length header
     let file_size = get_file_size(&response);
 
-    let filename = format!("{}.{}", title, extension);
+    let filename = format!("{}.{}", hq["title"], extension);
 
     // write file to disk
     write_file(response, &filename, file_size);
@@ -145,13 +136,11 @@ fn send_request(url: &str) -> Response {
     }
 }
 
-
 fn parse_url(query: &str) -> HashMap<String, String> {
     let u = format!("{}{}", "http://e.com?", query);
     let parsed_url = hyper::Url::parse(&u).unwrap();
     parsed_url.query_pairs().into_owned().collect()
 }
-
 
 fn read_line() -> String {
     let mut input = String::new();

--- a/src/main.rs
+++ b/src/main.rs
@@ -5,7 +5,7 @@ extern crate clap;
 extern crate regex;
 
 use pbr::ProgressBar;
-use std::str;
+use std::{process,str};
 use std::collections::HashMap;
 use hyper::client::response::Response;
 use hyper::Client;
@@ -45,7 +45,7 @@ fn download(url: &str) {
 
     if hq["status"] != "ok" {
         println!("Video not found!");
-        return;
+        process::exit(1);
     }
 
     // get video info
@@ -130,7 +130,10 @@ fn send_request(url: &str) -> Response {
     let ssl = NativeTlsClient::new().unwrap();
     let connector = HttpsConnector::new(ssl);
     let client = Client::with_connector(connector);
-    client.get(url).send().unwrap()
+    client.get(url).send().unwrap_or_else(|e| {
+        println!("Network request failed: {}", e);
+        process::exit(1);
+    })
 }
 
 fn parse_url(query: &str) -> HashMap<String, String> {

--- a/src/main.rs
+++ b/src/main.rs
@@ -18,7 +18,6 @@ use std::fs::File;
 use clap::{Arg, App};
 use regex::Regex;
 
-
 fn main() {
     //Regex for youtube URLs.
     let url_regex = Regex::new(r"^.*(?:(?:youtu\.be/|v/|vi/|u/w/|embed/)|(?:(?:watch)?\?v(?:i)?=|\&v(?:i)?=))([^#\&\?]*).*").unwrap();
@@ -34,7 +33,7 @@ fn main() {
         let vid_split = url_regex.captures(vid).unwrap();
         vid = vid_split.get(1).unwrap().as_str();
     }
-    let url = format!("http://youtube.com/get_video_info?video_id={}", vid);
+    let url = format!("https://youtube.com/get_video_info?video_id={}", vid);
     download(&url);
 }
 

